src/VisualStudio/Core/Impl/CodeModel/InternalElements/AbstractCodeElement.cs

```diff
@@ -56,9 +56,8 @@ protected ProjectId GetProjectId()
 
         internal bool IsValidNode()
         {
-            var node = LookupNode();
-
-            if (node == null)
+            SyntaxNode node;
+            if (!TryLookupNode(out node) || node == null)
```

Comment:
Shouldn't TryLookupNode return false if node is null.  THen you don't need the || check.